### PR TITLE
fix(oldfiles) : delete duplicated items in Windows

### LINKED ISSUE #1683: Duplicates path entries on Windows
### Description

There is duplication of entries when we are talking of paths, as seen in the screenshot below:
image

### Neovim version

```markdown
NVIM v0.7.0-dev+793-g991e872d8
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```

### Operating system and version

Windows 11

### checkhealth telescope

```markdown
telescope: require("telescope.health").check()
========================================================================
## Checking for required plugins
  - OK: plenary installed.
  - OK: nvim-treesitter installed.

## Checking external dependencies
  - OK: rg: found ripgrep 13.0.0 (rev af6b6c543b)
  - OK: fd: found fd 8.3.0

## ===== Installed extensions =====

## Telescope Extension: `projects`
  - INFO: No healthcheck provided
```

### Steps to reproduce

1. On windows, open a file a few times.
2. At some point it will duplicated in old_files, as there will be two path, one containing `\` the other having `\\`.

### Expected behavior

No duplication of paths, every folder/file have a single entry

### Actual behavior

Files and folders are sometimes duplicated due to having single or double forward slashes (as per screenshot in description)

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      {
        'nvim-telescope/telescope.nvim',
        requires = {
          'nvim-lua/plenary.nvim',
          { 'nvim-telescope/telescope-fzf-native.nvim', run = 'make' },
        },
      },
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
      display = { non_interactive = true },
    },
  }
end
_G.load_config = function()
  require('telescope').setup()
  require('telescope').load_extension('fzf')
  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Telescope and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
load_plugins()
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

### PR DESCRIPTION
**Pull request by Jaehaks**
# Description

1) Duplicated items are listed in oldfiles because of neovim problem.
   Example is like below.
```
   c:\Users\USERS\test\telescope.lua
   c:\Users\USERS/test/telescope.lua
   c:/Users/USERS/test/telescope.lua
```
   slash(/) makes neovim lua api cannot work properly in windows.
   and string comparison don't distinguish it is the same path

   This makes `file ~= current_file` condition cannot filter although it is same with the file

2) oldfiles show current session files in the list also.

Fixes #1683

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

1) Taking gsub('/', '\\') all files from neovim lua api. 
2) Add additional condition which checks whether the file is listed in results (='results_other') 

3) separate `results` variable to 2 parts.
    First, 'results' as current session file list
    Second, 'results_other' as old file list which are filtered unnecessary file

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list relevant details about your configuration

There is nothing for the user to configure.
- [x] At startup which has no opened buffer
- [x] Some buffers are opened
- [x] Load from saved session.   

**Configuration**:
* Neovim version (nvim --version): v0.9.5   and nvim-qt v0.2.17
* Operating system and version: Windows 10

# Checklist:

- [ ] My code follows the style guidelines of this project (stylua)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (lua annotations)

### CODE REVIEW

**Comment by Conni2461** [comment]
technically yes, seems like an easy fix, we can include this. but can you reduce the comments by like 95% we only have to mention it once why we do `gsub('/','\\')`.

and the formatting doesnt align with our style, this needs to be addressed as well

File: lua/telescope/builtin/__internal.lua [hunk 1]
```diff
@@ -525,30 +525,50 @@ end
 internal.oldfiles = function(opts)
   opts = apply_cwd_only_aliases(opts)
   opts.include_current_session = vim.F.if_nil(opts.include_current_session, true)
+  local has_win = vim.fn.has('win32') == 1
```
Comment (Conni2461) [inline lua/telescope/builtin/__internal.lua:528]: please use `utils.iswin()`

Comment (Jaehaks) [reply lua/telescope/builtin/__internal.lua:528]: OK I'll try it.

Comment (Jaehaks) [reply lua/telescope/builtin/__internal.lua]: > please use `utils.iswin()`

New PR is submitted

**Comment by jamestrew** [comment]
I have a fix on the plenary side for this + bunch of other Windows path issues https://github.com/nvim-lua/plenary.nvim/pull/618
This is a systemic issue so would rather fix it with plenary than with this PR.

**Comment by jamestrew** [comment]
Closed in favor of https://github.com/nvim-telescope/telescope.nvim/pull/3260

**Pull request closed without merging**